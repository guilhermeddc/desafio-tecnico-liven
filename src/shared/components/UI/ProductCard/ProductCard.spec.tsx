import {render, screen} from '@testing-library/react';
import {product01} from 'tests/mock';

import {ProductCard} from '.';

jest.mock('react-router-dom', () => ({
  useLocation: jest.fn(),
}));

describe('ProductCard', () => {
  it('renders correctly', () => {
    const {container} = render(<ProductCard product={product01} />);

    expect(container).toMatchSnapshot();
  });

  it('renders correctly with props', () => {
    render(<ProductCard product={product01} />);

    expect(screen.getByText(/Product Test 1/)).toBeInTheDocument();
    expect(screen.getByText(/10/)).toBeInTheDocument();
  });
});
