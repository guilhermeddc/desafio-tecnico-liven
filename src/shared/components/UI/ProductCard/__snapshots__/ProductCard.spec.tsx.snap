// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductCard renders correctly 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-4 css-63160j-MuiGrid-root"
    to="/product/1"
  >
    <div
      class="css-wgtysp-MuiStack-root"
    >
      <div
        aria-label="Imagem do produto"
        class="css-94a5y0-MuiStack-root"
      >
        <div
          class="css-18sz23g-MuiStack-root"
        >
          <h6
            aria-label="Product Test 1"
            class="MuiTypography-root MuiTypography-h6 MuiTypography-alignCenter css-1kgxqm0-MuiTypography-root"
          >
            Product Test 1
          </h6>
          <div
            class="css-fd7gp3-MuiStack-root"
          />
          <h4
            class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-1yae2gn-MuiTypography-root"
          >
            $10.00
          </h4>
        </div>
      </div>
    </div>
  </div>
</div>
`;
